tests/caasp/filesystem_ro.pm

```diff
@@ -14,17 +14,30 @@
 use base "opensusebasetest";
 use strict;
 use testapi;
+use version_utils 'is_caasp';
 
 sub run {
     assert_script_run "! touch /should_fail";
     assert_script_run "touch /etc/should_succeed";
     assert_script_run "touch /var/log/should_succeed";
 
     assert_script_run 'btrfs property get / ro | grep "ro=true"';
-    assert_script_run 'btrfs property get /var/log ro | grep "ro=false"';
 
-    assert_script_run "grep '/ btrfs ro' /etc/fstab";
-    assert_script_run "mount | grep 'on / type btrfs (ro,'";
+    if (is_caasp '4.0+') {
+        assert_script_run 'btrfs property get /var ro | grep "ro=false"';
+        assert_script_run 'lsattr -ld /var | grep No_COW';
+    }
+    else {
+        assert_script_run 'btrfs property get /var/log ro | grep "ro=false"';
+    }
+
+    if (is_caasp 'caasp') {
+        assert_script_run "grep '/ btrfs ro' /etc/fstab";
+        assert_script_run "mount | grep 'on / type btrfs (ro,'";
+    }
+    else {
+        record_soft_failure 'bsc#1079000 - Missing readonly option in fstab';
```

Comment:
So if I correctly understand the code, you record a softfailure if the product is 'caasp'? It's not more when the `assert_script_run "grep '/ btrfs ro' /etc/fstab";` test fails?

Reply:
Product is either caasp or kubic (opensuse version of caasp)
 - if it's caasp then I run checks
 - if it's kubic then I record soft failure

Reply:
Understood!